src/os/data/histo/colormodel.js

```diff
@@ -207,7 +237,8 @@ os.data.histo.ColorModel.prototype.setManualBinColors = function(colors) {
   goog.removeUid(colors);
 
   this.manualBinColors_ = colors;
-  this.applyColorMethod();
+
+  // this.applyColorMethod(); // already called by setColorMethod(), so don't double-up the re-draws for all the charts
```

Comment:
This may cause unwanted side effects in `restore`, where `setManualBinColors` is called last. Given the comment there about intentionally calling it last, it seems we can't just move that call. I am curious about that comment because if the color method isn't manual, there shouldn't be manual colors anyway (they would have been cleared when the color model was changed).

Reply:
I also moved "this.setColorMethod(...);" to the bottom in restore() for exactly that reason.  Is there a reason why it must be called before the setManualBinColors()?  In my testing I couldn't find any strange cases where it didn't load the manual bin colors properly from a restore with setColorMethod() last instead of setManualBinColors() last.

Reply:
Moving setColorMethod to the bottom was working for AUTO_COLOR, but had to add handler for MANUAL method.